spec/models/bounce_spec.rb

```diff
@@ -0,0 +1,201 @@
+
+    it "is true for blocked emails" do
+      expect(described_class.blocked?("blocked@example.org")).to be_truthy
+    end
+
+    it "is false for bounced, but unblocked emails" do
+      expect(described_class.blocked?("bounced@example.org")).to be_falsey
+    end
+
+    it "is false for unbounced emails" do
+      expect(described_class.blocked?("unbounced@example.com")).to be_falsey
+    end
+
+    it "is false for nil" do
+      expect(described_class.blocked?(nil)).to be_falsey
+    end
+  end
+
+  context "#person" do
+    it "finds the person by primary email" do
+      person = Fabricate(:person, email: "person@example.org")
+      bounce = described_class.create(email: person.email, count: 1)
+
+      expect(bounce.person).to eq(person)
+    end
+
+    it "finds the person by additional email" do
+      additional_email = "additional@example.org"
+      person = Fabricate(:person)
+      Fabricate(:additional_email, email: additional_email, contactable: person)
+
+      bounce = described_class.create(email: additional_email, count: 1)
+
+      expect(bounce.person).to eq(person)
+    end
+  end
+
+  context "#people" do
+    it "finds all people associated with that email-address" do
+      first_person = Fabricate(:person, email: "person@example.org")
+      second_person = Fabricate(:additional_email, email: "person@example.org").contactable
+
+      bounce = described_class.create(email: "person@example.org", count: 1)
+
+      expect(bounce.people).to match_array [first_person, second_person]
+    end
+  end
+
+  context "#people_ids" do
+    it "finds all people associated with that email-address" do
+      first_person_id = Fabricate(:person, email: "person@example.org").id
+      second_person_id = Fabricate(:additional_email, email: "person@example.org").contactable_id
+
+      bounce = described_class.create(email: "person@example.org", count: 1)
+
+      expect(bounce.people_ids).to match_array [first_person_id, second_person_id]
+    end
+  end
+
+  context "#mailing_lists" do
+    it "finds the mailing_lists that this email has bounce in" do
+      mailing_list = Fabricate(:mailing_list, group: groups(:top_layer))
+
+      bounce = described_class.record("person@example.org", mailing_list_id: mailing_list.id)
+
+      expect(bounce.mailing_lists).to match_array [mailing_list]
+    end
+  end
+
+  context "#block!" do
+    it "marks a bounce as blocked" do
+      expect do
+        bounced.block!
+      end.to change { bounced.reload.blocked? }.from(false).to(true)
+    end
+
+    it "returns the timestamp of blocking" do
+      expect do
+        expect(bounced.block!).to be_a DateTime
+      end.to change { bounced.reload.blocked_at }.from(nil)
+    end
+
+    context "of a blocked email" do
+      it "is accepted" do
+        expect do
+          blocked.block!
+        end.to_not change { blocked.reload.blocked? }
+      end
+
+      it "returns the timestamp of the original blocking" do
```

Comment:
Add a similar spec to check that `Bounce#record` also does not update blocked_at as soon as it has been blocked